### ISSUE TITLE
Can't debug dynamically loaded modules in node.js

### ISSUE THREAD

**Issue by maganap** [microsoft/vscode-js-debug#1084]
**Describe the bug**
When loading a module dynamically, all breakpoints inside that module are set as "Unbound breakpoint".
Breakpoints directly set on the program entry point (index.js) or any non-dynamically loaded module do work as expected.

**To Reproduce**
For X reason, I need to dynamically load some modules. I'm working this way:

```javascript
// ... inside some async function
  const reqPath = './fn/onCreateShippingLabels.js';
  await (await import(reqPath)).default(args);
// ... rest of code
```

Only breakpoints inside those modules show as "Unbound breakpoint".

**Log File**

`./fn/onCreateShippingLabels.js` is the actual module I'm trying to debug. It's being loaded from the entry point `index.js`.

[vscode-debugadapter-144f6919.json.gz](https://github.com/microsoft/vscode-js-debug/files/7023308/vscode-debugadapter-144f6919.json.gz)

**VS Code Version:** 
Version: 1.59.1
Commit: 3866c3553be8b268c8a7f8c0482c0c0177aa8bfa
Date: 2021-08-19T11:53:52.479Z
Electron: 13.1.7
Chrome: 91.0.4472.124
Node.js: 14.16.0
V8: 9.1.269.36-electron.0
OS: Darwin x64 19.6.0

I just installed `ms-vscode.js-debug-nightly` to test, but the same was happening with the previous `ms-vscode.js-debug` I had installed.

**Additional context**
This `launch.json` works just fine when loading modules the regular way:

```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Attach 1",
      "port": 9229,
      "request": "attach",
      "skipFiles": [
        "<node_internals>/**"
      ],
      "type": "pwa-node",
      "trace": true
    }
  ]
}
```

`package.json` includes:
```json
  "type": "module",
  "main": "index.js",
```

Is it a missing `launch.json` setting? Or is there another way to load modules dynamically for it to work? I've been googling for hours without luck.

Thank you very much in advance.

**Comment by connor4312** [comment]
In the log you attached I don't see `onCreateShippingLabels.js` getting loaded by the runtime at all, and actually `firebase2/functions/index.js` is the only script I see that gets loaded. I also can't repro with a simple example:

**index.js**

```js
(async () => {
  await (await import('./other.js')).default();
})();
```

**other.js**

```js
export default function() {
  console.log('hi'); // breakpoint here works
}
```

This may have something to do with how your scripts are being loaded or compiled.

**Comment by maganap** [comment]
@connor4312 You're absolutely right, it does work.  Thank you for your example, it helped me isolate the issue.

I did find something confusing, though:

When I started debugging it certainly didn't work. I was using modules but my entry point was `index.cjs`, not yet `index.js`. I was also invoking differently... I don't really remember the code structure, I was just testing, but to test I was 1) attaching the debugger and 2) invoking the function that dynamically loads the module.

I started changing one thing after the other trying to make the debugger work, until I opened this issue.

When you replied I realized that the breakpoints did work, but they were showing as "Unbound breakpoint" as soon as the debugger gets attached to the process. That's what got me confused.  I noticed that in my latest tests I was **only** attaching the debugger, but I was not invoking the function. So I'm not really sure about when it actually started working.

**Breakpoints would only show bound after invoking the function for the first time** (which makes sense I guess?). I wrongly assumed it wouldn't be working (as earlier) just because the breakpoints were not red.

I'll close this issue since that would be another topic of discussion. Thank you very much.

***********
Some reference pics of the confusion:

1) Process running, before attaching debugger:
![Screenshot 2021-08-22 at 11 20 56](https://user-images.githubusercontent.com/6942131/130350244-bdc83312-2ffd-41ef-ae77-5f69b1c93199.png)

2) after attaching debugger, function not invoked yet:
![Screenshot 2021-08-22 at 10 34 31](https://user-images.githubusercontent.com/6942131/130350276-3607a70f-abcc-4b09-8dcc-f67c1c7f4f4b.png)

3) After invoking the function for the first time, the breakpoints get bound and do work:
![Screenshot 2021-08-22 at 11 37 36](https://user-images.githubusercontent.com/6942131/130350285-f318e985-2c92-400f-befa-97257a822dde.png)
